scala-package/core/src/main/scala/ml/dmlc/mxnet/Optimizer.scala

```diff
@@ -71,6 +119,10 @@ abstract class Optimizer extends Serializable {
   // Dispose the state it created
   def disposeState(state: AnyRef): Unit
 
+  def serializeState(state: AnyRef): Array[Byte]
+
+  def deserializeState(bytes: Array[Byte]): AnyRef
```

Comment:
Haven't found a better way yet. I think it'll be too heavy to borrow generics here.